src/main/java/it/gov/pagopa/pu/registry/mapper/installment/DebtPositionIoEventDTO2InstallmentRegistryMapper.java

```diff
@@ -0,0 +1,44 @@
+package it.gov.pagopa.pu.registry.mapper.installment;
+
+import it.gov.pagopa.pu.registry.event.payments.dto.DebtPositionIoEventDTO;
+import it.gov.pagopa.pu.registry.model.InstallmentRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class DebtPositionIoEventDTO2InstallmentRegistryMapper {
+
+  public List<InstallmentRegistry> map(DebtPositionIoEventDTO dto) {
+    if (dto.getPayload().getMessages() == null) return List.of();
+
+    List<String> iudList;
+
+    if (dto.getEventDescription().startsWith("IUD:")) {
```

Comment:
probably you will have many events which should perform iud extraction, thus I would create an utility method to perform this once for all.

I would make it more reusable:
* `IUD:` could not be at the start, just check if it's contained
* Let's return a Set instead of List
* Not assuming anymore IUD at the start of the string, before to extract the iud list, you should substring the description at index indexOf("IUD:")+5, take care that, using substring and the moving the index of 5 element we should move after the `:` charater, thus the split operation should not more be necessary

we could add a terminal `;` if you want, in order to standardize the delimiter in case of many parameters in this field